pkg/store/prometheus.go

```diff
@@ -496,9 +499,39 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		sers []map[string]string
+		err  error
+	)
+
+	vals := []string{}
+	version := p.promVersion()
+
+	if len(r.Matchers) == 0 || version > "2.24" {
+		vals, err = p.client.LabelValuesInGRPC(ctx, p.base, r.Label, r.Matchers, r.Start, r.End)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		matchers, err := storepb.MatchersToPromMatchers(r.Matchers...)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		sers, err = p.client.SeriesInGRPC(ctx, p.base, matchers, r.Start, r.End)
+		if err != nil {
+			return nil, err
+		}
+
+		// Using set to handle duplicate values.
```

Comment:
Although we intend a `set` behaviour, we are using `map` right? Lets change it to `map`. WDYT?
```suggestion
		// Using map to handle duplicate values.
```

Reply:
> Although we intend a set behaviour, we are using map right?

Yes
So should we change it to `Using map to deduplicate values` ?

Reply:
We can keep the comment as map[string]struct{} is a common pattern in go for a set